samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationServiceFactory.java

```diff
@@ -19,32 +19,57 @@
 package org.apache.samza.zk;
 
 import com.google.common.base.Strings;
+import java.util.Map;
 import org.I0Itec.zkclient.ZkClient;
 import org.apache.samza.SamzaException;
 import org.apache.samza.config.Config;
+import org.apache.samza.config.MetricsConfig;
 import org.apache.samza.config.ZkConfig;
 import org.apache.samza.coordinator.CoordinationServiceFactory;
 import org.apache.samza.coordinator.CoordinationUtils;
+import org.apache.samza.metrics.MetricsRegistryMap;
+import org.apache.samza.metrics.MetricsReporter;
+import org.apache.samza.util.MetricsReporterLoader;
 import org.apache.zookeeper.client.ConnectStringParser;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 public class ZkCoordinationServiceFactory implements CoordinationServiceFactory {
-  private final static Logger LOG = LoggerFactory.getLogger(ZkCoordinationServiceFactory.class);
+  private static final Logger LOG = LoggerFactory.getLogger(ZkCoordinationServiceFactory.class);
+
+  private static final String SOURCE_PREFIX = "ZkJobCoordinationService";
 
   public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
     ZkConfig zkConfig = new ZkConfig(config);
 
     ZkClient zkClient =
         createZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
 
-    ZkUtils zkUtils = new ZkUtils(new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs());
+    ZkJobCoordinatorMetrics zkJobCoordinatorMetrics = getJobCoordinatorMetrics(config, participantId);
+
+    ZkUtils zkUtils = new ZkUtils(new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs(), zkJobCoordinatorMetrics);
```

Comment:
I am not opposed to having this `isLeader` metric or any metric for CoordinationService. Here are the reasons why I think the solution provided here won't work or is not comprehensive:
1. Merely creating the metrics reporter and registering the metrics will not actually report any metrics. We need to start/stop the metrics reporter , aka manage its lifecycle. Until we do that, we really don't have any metrics.
2. I just noticed this today. The `participantId` used in ZkCoordinationService is a random UUID. I suspect that is because of the changes from SEP-1 that CoordinationService does not have access to the processorId. So, in reality, we will be getting metrics around a whole bunch of different *processorIds*. I am not sure if you can actually associate a temporary uuid with any processor itself. Please take a look at [LocalApplicationRunner Line 134](https://github.com/apache/samza/blob/4875842b37e0285100e2d1d753a7bc4a1448e897/samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java#L134)